agent/run_job.go

```diff
@@ -166,6 +161,87 @@ func (r *JobRunner) Run(ctx context.Context) error {
 	return nil
 }
 
+func (r *JobRunner) validateConfigAllowlists(job *api.Job) error {
+	validations := map[string]func() error{
+		"repo": func() error {
+			return validateJobValue(r.conf.AgentConfiguration.AllowedRepositories, job.Env["BUILDKITE_REPO"])
+		},
+		"environment variables": func() error {
+			return validateEnv(job.Env, r.conf.AgentConfiguration.AllowedEnvironmentVariables)
+		},
+		"plugins": func() error {
+			return r.validatePlugins()
+		},
+	}
+
+	for name, validation := range validations {
+		if err := validation(); err != nil {
+			return fmt.Errorf("failed to validate %s: %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+// validateEnv checks if the environment variables are allowed by ensuring their names match at least one of the regular
+// expressions in the allowedVariablePatterns list. The error message will contain all of the variables that aren't are
```

Comment:
"The error message will contain all of the variables that aren't are invalid." 

Grammar-o aside, is that true?

Reply:
ahhh not any more! i will make it true again